File: Front/TS/src/infrastructure/adapters/AddFriendPresenter.ts
import { IAddFriendOutputPort } from "../../domain/features/add-friend/AddFriendHandler";
import { Subject, Observer } from 'rxjs';

export class AddFriendPresenter implements IAddFriendOutputPort {
    private friendSubject = new Subject<FriendAdded>();
    Present(memberId: string): void {
        this.friendSubject.next({MemberId: memberId})
    }

    Subscribe(observer: Observer<FriendAdded>) {
        this.friendSubject.subscribe(observer)
    }
}

export interface FriendAdded {
    MemberId: string;
}

File: Front/.Net Maui/BetFriends.Blazor/Presenters/AddFriendPresenter.cs
﻿using BetFriend.Domain.Features.AddFriend;
using CommunityToolkit.Mvvm.Messaging;

namespace BetFriends.Blazor.Presenters;

internal class AddFriendPresenter : IAddFriendOutputPort
{
    public void Present(string id)
    {
        WeakReferenceMessenger.Default.Send(new FriendAdded(id));
    }
}

internal record FriendAdded(string Id);